server/pulp/server/async/app.py

```diff
@@ -49,6 +61,27 @@ def initialize_worker(sender, instance, **kwargs):
     """
     initialization.initialize()
 
+    _logged = False
+
+    if sender.startswith(constants.RESOURCE_MANAGER_WORKER_NAME):
+
+        name = constants.RESOURCE_MANAGER_WORKER_NAME + "@" + platform.node()
+        lock = ResourceManagerLock(name=name)
+
+        while True:
+            try:
+                lock.save()
+                break
+            except mongoengine.NotUniqueError:
+                if not _logged:
+                    msg = _("An instance of pulp_resource_manager is already running. "
```

Comment:
The message should probably be more along the lines of "An instance of pulp_resource_manager is already active. Halting execution until the currently active resource_manager becomes unavailable."